docs/stacks/index.md

```diff
@@ -13,30 +13,63 @@ next:
 
 # Stack Configuration
 
-When working with Infrastructure as Code it's considered to be a best practice
-to split up and organize your IaC into several smaller and isolated stacks.
+When working with Infrastructure as Code (IaC), adopting a modular approach is highly recommended. This approach breaks the entire IaC into smaller, isolated **stacks**, enabling code and infrastructure component reuse. 
 
-Typically, each stack comes with its own Terraform state which allows us
-to plan and apply each stack on its own.
+Additionally, it enhances infrastructure management across multiple stacks and facilitates testing and deploying changes in a controlled manner, minimizing unintended consequences.
 
-A Terramate stack is:
+The Terramate CLI stack is a powerful feature of the Terramate CLI tool, allowing you to manage complex deployments with ease.
 
-* A directory inside your project.
-* Has at least one or more Terramate configuration files.
-* One of the configuration files has a `stack {}` block on it.
 
-What separates a stack from any other directory is the `stack{}` block.
-It doesn't require any attributes by default, but it can be used
-to describe stacks and orchestrate their execution.
+## What is a Stack?
+
+A stack is a collection of related resources managed as a single unit. When defining a stack, you specify all included resources, such as networks, virtual machines, and storage. 
+
+Terramate CLI simplifies resource creation and management, allowing you to focus on building and deploying the application.
+
+
+## A Terramate Stack is:
+
+- A directory inside your project
+- Contains one or more Terramate configuration files
+- Includes a configuration file with a stack{} block
+
+The `stack{}` block distinguishes a stack from other directories in Terramate. By default, it doesn't require any attributes but can be used to describe stacks and orchestrate their execution.
 
 Stack configurations related to orchestration can be found [here](../orchestration/index.md).
 
-Besides orchestration the `stack` block also define attributes that are
-used to describe the `stack`.
-
-Only [Terramate Functions](../functions/index.md) are available when defining
+The `stack{}` block also defines attributes used to describe the stack. Only [Terramate Functions](../functions/index.md) are available when defining
 the `stack` block.
 
+## Why use Stacks?
+
+Stacks offer several benefits:
+
+1. Manage multiple resources as one unit, allowing you to build and deploy the entire infrastructure efficiently and quickly with a single command.
+
+2. Easily manage dependencies between resources. For example, you can specify a virtual machine that depends on a virtual network, and Terramate CLI will automatically manage the resource's creation or deletion.
```

Comment:
this part `Terramate CLI will automatically manage the resource's creation or deletion.` is not entirely correct... This could be misleading for new users because they could think that Terramate does some kind of automatic resource management.